mangaki/mangaki/utils/knn.py

```diff
@@ -118,20 +87,29 @@ def fit(self, X=None, y=None, whole_dataset=False):
             self.M[user_id, work_id] = rating
         for work_id in self.nb_ratings:
             self.mean_score[work_id] = self.sum_ratings[work_id] / self.nb_ratings[work_id]
-        self.get_all_neighbors()
 
     def predict(self, X):
+        self.get_neighbors(list(set(X[:, 0])))  # Compute only relevant neighbors
```

Comment:
Because I need to provide a list, else it is impossible to slice the matrix later on.